### DISCUSSION TITLE
bot not working after a while

### DISCUSSION THREAD

**Discussion by Cardboard-Friend** [eritislami/evobot#987] [General]
after a few hours of bot being up I get this error when trying to play music:
05.10 17:38:30 [Bot] RuntimeError: input stream: memory access out of bounds
05.10 17:38:30 [Bot] at wasm-function[268]:0x2177a
05.10 17:38:30 [Bot] at wasm-function[267]:0x21732
05.10 17:38:30 [Bot] at OpusScriptHandler$_decode [as _decode] (eval at Db (/node_modules/opusscript/build/opusscript_native_wasm.js:1:1), <anonymous>:9:10)
05.10 17:38:30 [Bot] at OpusScript.decode (/node_modules/opusscript/index.js:80:28)
05.10 17:38:30 [Bot] at Decoder._decode (/node_modules/prism-media/src/opus/Opus.js:64:25)
05.10 17:38:30 [Bot] at Decoder._transform (/node_modules/prism-media/src/opus/Opus.js:204:22)
05.10 17:38:30 [Bot] at Decoder.Transform._read (_stream_transform.js:191:10)
05.10 17:38:30 [Bot] at Decoder.Transform._write (_stream_transform.js:179:12)
05.10 17:38:30 [Bot] at doWrite (_stream_writable.js:403:12)
05.10 17:38:30 [Bot] at writeOrBuffer (_stream_writable.js:387:5)

**Comment by drewburr** [comment]
@Bruciemm could you provide more information about your setup?

1. Which version of Evobot are you using?
2. Are you running the bot using Docker or NodeJS?
3. Are you able to reproduce this reliably? What steps can we take to reproduce this issue?
4. How much memory is available on the machine you're running the bot on?

**Comment by Cardboard-Friend** [comment]
1. latest version
2. notejs
3. I cannot reproduce it, it just happens after a while
4. 1024 MB, using pebble host

**Comment by eritislami** [comment]
https://pm2.keymetrics.io/
pm2 start index.js --max-memory-restart 300M

**Reply by Cardboard-Friend** [reply]
how exactly do i install pm2, the bot host doesnt give access to the server console to install the way it says on website, do i have to edit a file or something?

**Reply by Cardboard-Friend** [reply]
just checked there is no way to install pm2 as its just a bot hosting server and I have no access to root system

**Reply by eritislami** [reply]
> just checked there is no way to install pm2 as its just a bot hosting server and I have no access to root system

Well its not my fault if your host has memory issues

**Reply by wolf-yuan-6115** [reply]
For some host (eg. Heroku) need to start process with `pm2-runtime`
Please try:
`pm2-runtime start index.js --max-memory-restart 300M`

**Reply by Cardboard-Friend** [reply]
do you have any bot host suggestions?

**Reply by MujiSayed** [reply]
@Bruciemm I'd recommend getting a raspberry pi and setting it up on that. The new raspberry pi's have 4gb/8gb ram, and what i'm running my bot on
